runtime/src/accounts_background_service.rs

```diff
@@ -360,20 +365,27 @@ impl SnapshotRequestHandler {
                     self.snapshot_config.snapshot_version,
                 )
                 .expect("snapshot bank");
-                AccountsPackage::new_for_snapshot(
+                let accounts_package = AccountsPackage::new_for_snapshot(
                     accounts_package_type,
                     &snapshot_root_bank,
                     &bank_snapshot_info,
                     &self.snapshot_config.bank_snapshots_dir,
                     status_cache_slot_deltas,
                     &self.snapshot_config.full_snapshot_archives_dir,
                     &self.snapshot_config.incremental_snapshot_archives_dir,
-                    snapshot_storages,
+                    snapshot_storages.clone(),
                     self.snapshot_config.archive_format,
                     self.snapshot_config.snapshot_version,
                     accounts_hash_for_testing,
                 )
-                .expect("new accounts package for snapshot")
+                .expect("new accounts package for snapshot");
+                snapshot_slot_storages.push_back(snapshot_storages);
+
+                // Remove the older ones, causing them to release the reference counts of the appendvecs
+                while snapshot_slot_storages.len() > 1 {
+                    snapshot_slot_storages.pop_front();
+                }
```

Comment:
I will reply inline below.

> A few thoughts here:
> 
> * Can the name of the `snapshot_slots_storages` change to `last_snapshot_storages` so that it mirrors `last_full_snapshot_slot`?

at this low level, I intend to make the function focus on capturing memory data into the file system, not having the any concern about whether it is the latest.  It is better to let the upper app level to care about which one is latest.


> * If we only keep the most recent `snapshot_storages`, why even keep a list around at all? Instead, just set the latest one directly. Then this would be `last_snapshot_storages = snapshot_storages` and all the `VecDeque` stuff can go away.

Agreed.  Will remove the VecDeque.

> * In `AccountsBackgroundService::new()`, it takes in a `last_full_snapshot_slot` parameter since that's computed at startup. Should something similar happen here for the snapshot storages? Specifically, what happens in the case where we load from a snapshot (archive or otherwise) but then stop before the next bank snapshot is taken? In that scenario, is it possible to still have a fastboot from the accounts already extracted on disk? I'm thinking currently the answer is "no", but can/should it be "yes" later?
Will use the file flag to signal the completeness of the directory content.  The partial directory content without the completion flag should not be used for constructing a bank.

> * The `last_snapshot_storages` doesn't need to be a parameter to this function either, since it's never _read_. I think it'd be cleaner to _return_ `snapshot_storages` from this function and then update `last_snapshot_storages`  in `AccountsBackgroundService::new()` where `handle_snapshot_requests()` is called.

There are two levels of calls: handle_snapshot_requests and handle_snapshot_request from the top level loop.  Passing it back via return value requires wrapping it into a tuple with another return value, for both functions.  It does look to be clean.  I think it is better to leave it this way, or put it into the struct SnapshotRequestHandler.  @brooksprumo, please let me know your preference.  Thanks!

Reply:
The code is updated to take the suggestions.  Thanks!